mm2src/lp_ordermatch/simple_market_maker.rs

```diff
@@ -50,6 +48,13 @@ pub enum OrderProcessingError {
     BalanceIsZero,
     #[display(fmt = "Error when creating the order")]
     OrderCreationError,
+
+    AssetError,
+    LegacyError(String),
+}
+
+impl From<std::string::String> for OrderProcessingError {
+    fn from(error: String) -> Self { OrderProcessingError::LegacyError(error) }
 }
```

Comment:
Not a big fan of such traits, hard to understand which kind of error we catch, but I guess we have no choice if we treat legacy error string.